extensions/ql-vscode/src/data-extensions-editor/extension-packs.ts

```diff
@@ -66,27 +73,107 @@ export async function pickModelFile(
     }
   }
 
-  const fileOptions: Array<{ label: string; file: string }> = [];
+  const fileOptions: Array<{ label: string; file: string | null }> = [];
   for (const file of modelFiles) {
     fileOptions.push({
       label: relative(extensionPackPath, file),
       file,
     });
   }
+  fileOptions.push({
+    label: "Create new model file",
+    file: null,
+  });
 
   progress({
     message: "Choosing model file...",
     step: 3,
     maxStep: 3,
   });
 
-  const fileOption = await window.showQuickPick(fileOptions, {
-    title: "Select model file to use",
-  });
+  const fileOption = await window.showQuickPick(
+    fileOptions,
+    {
+      title: "Select model file to use",
+    },
+    token,
+  );
 
   if (!fileOption) {
-    return;
+    return undefined;
+  }
+
+  if (fileOption.file) {
+    return fileOption.file;
+  }
+
+  return pickNewModelFile(token, databaseItem, extensionPackPath);
+}
+
+async function pickNewModelFile(
+  token: CancellationToken,
+  databaseItem: DatabaseItem,
+  extensionPackPath: string,
+) {
+  const qlpackPath = await getQlPackPath(extensionPackPath);
+  if (!qlpackPath) {
+    void showAndLogErrorMessage(
+      `Could not find any of ${QLPACK_FILENAMES.join(
+        ", ",
+      )} in ${extensionPackPath}`,
+    );
+    return undefined;
+  }
+
+  const qlpack = await loadYaml(await readFile(qlpackPath, "utf8"), {
+    filename: qlpackPath,
+  });
+  if (typeof qlpack !== "object" || qlpack === null) {
+    void showAndLogErrorMessage(`Could not parse ${qlpackPath}`);
+    return undefined;
+  }
+
+  const dataExtensionPatterns = qlpack.dataExtensions ?? [];
+  if (!Array.isArray(dataExtensionPatterns)) {
+    void showAndLogErrorMessage(
+      `Expected 'dataExtensions' to be an array in ${qlpackPath}`,
+    );
```

Comment:
This is not correct. The following are legal and semantically identical:

```
dataExtensions:
  - "*/foo.yml"
```

and

```
dataExtensions: "*/foo.yml"
```

It is a quirk of the Java yaml parsing library we are using.